crates/router/src/scheduler/workflows/create_auto_refund.rs

```diff
@@ -0,0 +1,126 @@
+use api_models::{
+    enums::EventType,
+    refunds::{RefundRequest, RefundType},
+    webhooks::OutgoingWebhookContent,
+};
+use common_utils::ext_traits::ValueExt;
+use diesel_models::{
+    enums::{EventClass, EventObjectType},
+    refund,
+};
+
+use super::{AutoRefundWorkflow, ProcessTrackerWorkflow};
+use crate::{
+    core::{
+        errors::ApiErrorResponse,
+        refunds::{add_auto_refund_task_to_process_tracker, refund_create_core},
+        webhooks::create_event_and_trigger_appropriate_outgoing_webhook,
+    },
+    errors,
+    logger::error,
+    routes::AppState,
+    types::storage,
+};
+
+#[async_trait::async_trait]
+impl ProcessTrackerWorkflow for AutoRefundWorkflow {
+    async fn execute_workflow<'a>(
+        &'a self,
+        state: &'a AppState,
+        process: storage::ProcessTracker,
+    ) -> Result<(), errors::ProcessTrackerError> {
+        let db = &*state.store;
+        let tracking_data: refund::AutoRefundWorkflow = process
+            .tracking_data
+            .clone()
+            .parse_value("AutoRefundWorkflow")?;
+        let payment_intent = tracking_data.payment_intent;
+        let retry_count = tracking_data.retry_count;
+        let max_retries = tracking_data.max_retries;
+        if retry_count > max_retries {
+            return Err(errors::ProcessTrackerError::FlowExecutionError {
+                flow: "RefundCreate",
+            });
+        }
+        let key_store = state
+            .store
+            .get_merchant_key_store_by_merchant_id(
+                payment_intent.clone().merchant_id.as_str(),
+                &state.store.get_master_key().to_vec().into(),
+            )
+            .await?;
+        let merchant_account = db
+            .find_merchant_account_by_merchant_id(
+                payment_intent.clone().merchant_id.as_str(),
+                &key_store,
+            )
+            .await?;
+        let ref_req = RefundRequest {
+            refund_id: None,
+            payment_id: payment_intent.clone().payment_id,
+            merchant_id: Some(payment_intent.clone().merchant_id),
+            amount: None,
+            reason: Some("Auto Refund".to_string()),
+            refund_type: Some(RefundType::Scheduled),
+            metadata: None,
+            merchant_connector_details: None,
+        };
+        let refund_flow_result =
+            refund_create_core(state, merchant_account.clone(), key_store, ref_req).await;
```

Comment:
`refund_create_core` creates refund for the `last_successful_attempt` so if there is a previous attempt for which refund should be created , a different function needs to be used.

Reply:
Okay.